lib/widgets/recent_dm_conversations.dart

```diff
@@ -138,3 +149,55 @@ class RecentDmConversationsItem extends StatelessWidget {
           ]))));
   }
 }
+
+class _NewDmButton extends StatefulWidget {
+  final VoidCallback onPressed;
+
+  const _NewDmButton({required this.onPressed});
+
+  @override
+  State<_NewDmButton> createState() => _NewDmButtonState();
+}
+
+class _NewDmButtonState extends State<_NewDmButton> {
+  bool _pressed = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final fabBgColor = _pressed ? designVariables.fabBgPressed : designVariables.fabBg;
+    final fabLabelColor = _pressed ? designVariables.fabLabelPressed : designVariables.fabLabel;
```

Comment:
nit: these lines are too long; wrap them so that they stay under 80 characters in width